subxt/src/client/online_client.rs

```diff
@@ -209,34 +233,110 @@ impl<T: Config> ClientRuntimeUpdater<T> {
         &curr.runtime_version != new
     }
 
+    fn do_update(&self, update: Update) {
+        let mut writable = self.0.inner.write();
+        writable.metadata = update.metadata;
+        writable.runtime_version = update.runtime_version;
+    }
+
+    /// Tries to apply a new update.
+    pub fn apply_update(&self, update: Update) -> Result<(), UpgradeError> {
+        if !self.is_runtime_version_different(&update.runtime_version) {
+            return Err(UpgradeError::SameVersion)
+        }
+
+        self.do_update(update);
+
+        Ok(())
+    }
+
     /// Performs runtime updates indefinitely unless encountering an error.
     ///
     /// *Note:* This will run indefinitely until it errors, so the typical usage
     /// would be to run it in a separate background task.
```

Comment:
Sorry; I meant this comment that was now a bit out of date :)

Reply:
yeah but it's still true.

I mean it's depends whether we regard `failed runtime upgrades` as error before we didn't :P

Reply:
Oh actually I completely missed a `?` in the while loop in the new one, so I see what you mean; my bad :)